core/src/banking_stage.rs

```diff
@@ -394,21 +395,25 @@ impl BankingStage {
             block_production_method,
             BlockProductionMethod::CentralSchedulerGreedy
         );
-        let (non_vote_exit_signal, non_vote_thread_hdls) = Self::new_central_scheduler(
+        let non_vote_context = BankingStageNonVoteContext {
+            non_vote_exit_signal: Arc::new(AtomicBool::new(false)),
+            non_vote_receiver,
+            transaction_recorder,
+            poh_recorder,
+            bank_forks,
+            committer,
+            log_messages_bytes_limit,
+        };
+        let non_vote_thread_hdls = Self::new_central_scheduler(
             transaction_struct,
             use_greedy_scheduler,
             num_workers,
-            non_vote_receiver,
-            transaction_recorder.clone(),
-            poh_recorder.clone(),
-            bank_forks.clone(),
-            committer.clone(),
-            log_messages_bytes_limit,
+            non_vote_context.clone(),
```

Comment:
Looks good overall, just curious if considered to have `BankingStage` owning `Option<Arc< BankingStageNonVoteContext>>`, so here `Arc<...>` is cloned instead of `BankingStageNonVoteContext`.

Reply:
In the next PR it'll become an `Arc<RwLock<Option<BankingStageNonVoteContext>>`. but that wrapping didn't seem useful during this PR.